src/libs/permission.ts

```diff
@@ -0,0 +1,13 @@
+function hasPermission(moduleName, role, permissionType) {
+    if (!moduleName.hasOwnProperty(permissionType)) {
+        return false;
+    }
+    else if (moduleName[permissionType].includes(role) || role === 'head-trainer') {
```

Comment:
You should not have explicit check on role here, it would be better if you check based on "all" permission type. In your current implementation if you add another role that also has all permission rights than you need to change this file and the constants file as well. 

Also you are passing moduleName as object it should be a string.